### DISCUSSION TITLE
Predefine arrow connections between connectors without Users Drag action

### DISCUSSION THREAD

**Discussion by Klausero** [eckig/graph-editor#122] [Q&A] [answered]
Hello Eckig, I am trying to set a method to pre-build connections between nodes instead of creating them in the Editor. The idea is avoid any user Drag and drop actions, but instead draw the connections between nodes when the GraphEditorContainer is shown in the screen:

![grafik](https://user-images.githubusercontent.com/107101963/202661456-755a0335-2b8a-4464-8608-801643f9b544.png)

The idea is build such connections by crossing data from the Database, which the relations/connections between nodes are defined in its corresponding tables. As for Nodes, Connectors can also have Ids that could be potentially linked to each other (outputConnectorId -> inputConnectorId), but I cannot find the command or an example where the user action is actually translated into this command. Is there an example for this or could you please let me know which command works for this action?

Edit: I found the GConnection class with the implementation (GConnectionClass), so I thought I could do my own customized connection class, but the GModel does not have a method for setConnections

**Comment by eckig** [comment, marked as answer]
Have a look into the wiki: https://github.com/eckig/graph-editor/wiki/2.-Feature-Overview

You have the `GNode`s in the model. Each node has `GConnector`s and each connector has `GConnection`s. So you can add the connections programmatically by adding them to the connectors.

You can also look into this issue, where a similar question was asked: https://github.com/eckig/graph-editor/issues/42

**Reply by Klausero** [reply]
Thank you, I will have a look :)
